sensor/debugger/k8s/fake.go

```diff
@@ -0,0 +1,271 @@
+		networkPolicyKind: func(namespace string) interface{} {
+			return f.Client.Kubernetes().NetworkingV1().NetworkPolicies(namespace)
+		},
+		serviceKind:    func(namespace string) interface{} { return f.Client.Kubernetes().CoreV1().Services(namespace) },
+		jobKind:        func(namespace string) interface{} { return f.Client.Kubernetes().BatchV1().Jobs(namespace) },
+		replicaSetKind: func(namespace string) interface{} { return f.Client.Kubernetes().AppsV1().ReplicaSets(namespace) },
+		replicationControllerKind: func(namespace string) interface{} {
+			return f.Client.Kubernetes().CoreV1().ReplicationControllers(namespace)
+		},
+		daemonSetKind:   func(namespace string) interface{} { return f.Client.Kubernetes().AppsV1().DaemonSets(namespace) },
+		deploymentKind:  func(namespace string) interface{} { return f.Client.Kubernetes().AppsV1().Deployments(namespace) },
+		statefulSetKind: func(namespace string) interface{} { return f.Client.Kubernetes().AppsV1().StatefulSets(namespace) },
+		cronJobKind:     func(namespace string) interface{} { return f.Client.Kubernetes().BatchV1().CronJobs(namespace) },
+		podKind:         func(namespace string) interface{} { return f.Client.Kubernetes().CoreV1().Pods(namespace) },
+	}
+	f.resourceMap = map[string]interface{}{
+		namespaceKind:             &corev1.Namespace{},
+		secretKind:                &corev1.Secret{},
+		serviceAccountsKind:       &corev1.ServiceAccount{},
+		roleKind:                  &rbacv1.Role{},
+		clusterRoleKind:           &rbacv1.ClusterRole{},
+		roleBindingKind:           &rbacv1.RoleBinding{},
+		clusterRoleBindingKind:    &rbacv1.ClusterRoleBinding{},
+		networkPolicyKind:         &networkingv1.NetworkPolicy{},
+		nodeKind:                  &corev1.Node{},
+		serviceKind:               &corev1.Service{},
+		jobKind:                   &batchv1.Job{},
+		replicaSetKind:            &appsv1.ReplicaSet{},
+		replicationControllerKind: &corev1.ReplicationController{},
+		daemonSetKind:             &appsv1.DaemonSet{},
+		deploymentKind:            &appsv1.Deployment{},
+		statefulSetKind:           &appsv1.StatefulSet{},
+		cronJobKind:               &batchv1.CronJob{},
+		podKind:                   &corev1.Pod{},
+	}
+}
+
+// waitForMinimumResources waits for a minimum number of resources to be created or once all the events have been processed
+func waitForMinimumResources(ch chan string, done concurrency.Signal) error {
+	count := 0
+	doneC := done.WaitC()
+	for {
+		select {
+		case obj := <-ch:
+			if _, ok := minimumResources[obj]; ok {
+				minimumResources[obj]--
+				if minimumResources[obj] == 0 {
+					count++
+				}
+				if len(minimumResources) == count {
+					return nil
+				}
+			}
+		case <-doneC:
+			return errors.New("the events file did not contain the minimum resources required to start sensor")
+		}
+	}
+}
+
+// CreateEvents creates the k8s events from a given jsonl file
+func (f *FakeEventsManager) CreateEvents() error {
+	ch := make(chan string)
```

Comment:
(read this comment after reading my other comments) 

Here we have the following possibilities:
1. Close the channel as suggested and see whether there are any issues. If any panics appear we need to then implement the concurrency patterns as I suggested in other comments.
2. Not close the channel and leave it as it is. In this particular case there should be no severe consequences of this. I think it might be fine leaving it like this as this coat will not be run in the production.

Reply:
I think we can just close the channel in the `CreateEvents` function. 

```golang
ch := make(chan string)
f.done = concurrency.NewSignal()
objs, err := f.Reader.ReadFile()
if err != nil {
    return err
}
f.Init()
go func() {
    defer close(ch)
```

Reply:
Just to keep record here: After our chat we reworked all the concurrency and it is significantly different now.